src/USDA.NET/BaseClient.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+
+namespace USDA.NET
+{
+    public class BaseClient : IDisposable
+    {
+        protected readonly HttpClient HTTPClient;
+        protected List<KeyValuePair<string, string>> ParameterDictionary { get; set; } = new List<KeyValuePair<string, string>>();
+
+        protected BaseClient(string apiKey)
+        {
+            HTTPClient = new HttpClient
+            {
+                BaseAddress = new Uri("https://api.nal.usda.gov")
+            };
+
+            ParameterDictionary.Add(new KeyValuePair<string, string>("format", "json"));
+            ParameterDictionary.Add(new KeyValuePair<string, string>("api_key", apiKey));
+        }
+
+        public void Dispose()
+        {
+            HTTPClient?.Dispose();
```

Comment:
`HTTPClient` should not be null here, making the null propagation unnecessary. You can protect yourself by making the setter of `HTTPClient` private and the getter protected.

Reply:
the property is already a protected readonly so that should be the same thing right? I can remove the null propagation still if you'd like. Theoretically, it can't be null since it's init'ed in the ctor.